highlighters/main/main-highlighter.zsh

```diff
@@ -442,16 +537,23 @@ _zsh_highlight_highlighter_main_paint()
     # Parse the sudo command line
     if (( ! in_redirection )); then
       if [[ $this_word == *':sudo_opt:'* ]]; then
-        case "$arg" in
+        if [[ -n $flags_with_argument ]] &&
+           [[ $arg == '-'[$flags_sans_argument]#[$flags_with_argument] ]]; then
           # Flag that requires an argument
-          '-'[Cgprtu]) this_word=${this_word//:start:/};
-                       next_word=':sudo_arg:';;
+          this_word=${this_word//:start:/}
+          next_word=':sudo_arg:'
+        elif [[ $arg == '-'* ]]; then
+          # Flag that requires no argument, or unknown flag.
           # This prevents misbehavior with sudo -u -otherargument
-          '-'*)        this_word=${this_word//:start:/};
-                       next_word+=':start:';
-                       next_word+=':sudo_opt:';;
-          *)           ;;
-        esac
+          this_word=${this_word//:start:/}
+          next_word+=':start:'
+          next_word+=':sudo_opt:'
```

Comment:
Perhaps to be pedantically correct ```(( $#precommand_options[$arg] )) && this_word+=':sudo_opt:'```.

Reply:
What about `nice -x ls` where -x is an argumentless option not known to `precommand_options`?

Reply:
I don't know if we should assume all unknown options are argumentless. I think it'd be better to fail hard (-x is likely not an executable and so is unknown-token), so someone can either report it upstream or locally extend their `precommand_options`

Reply:
This is appealing: it means the first unknown -x will be highlighted as an error regardless of whether or not it takes an argument.

I'll make it so.

Reply:
Made so in e27e53b.